xonsh/history/base.py

```diff
@@ -153,3 +154,21 @@ def run_gc(self, size=None, blocking=True):
             If set blocking, then wait until gc action finished.
         """
         pass
+
+    @staticmethod
+    def remember_history_check(f):
```

Comment:
Why does this decorator need to exist? It seems easier/faster to put the logic in the actual functions where it is needed. All this is doing is a simple if-condition.

Reply:
It doesn't need to exist. In my opinion it's cleaner/more readable to use the decorator, but I'm happy to move the check into the individual functions if that's what you prefer.

Reply:
Yeah, please move them. I am pretty sure that this implementation is slower.